### PR TITLE
feat: heroku button

### PR DESCRIPTION
**Pull request by tonylyjones**
## Summary by CodeRabbit

- **New Features**
  - Added detailed instructions and a button link for deploying Azimutt on Heroku in the documentation.
  - Introduced configuration settings in `app.json` for Heroku deployment, including environment variables and add-ons.
  - Included a `heroku.yml` file specifying Dockerfile usage for building the web component.
  - Added a post-deployment script for setting environment variables on Heroku.

- **Documentation**
  - Reorganized and updated the local development setup information in the README.
  
- **Configuration**
  - Enhanced S3 credentials retrieval in runtime configuration with additional environment variables.

### CODE REVIEW

File: app.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    "database schema",
+    "diagram"
+  ],
+  "stack": "container",
+  "addons": [{ "plan": "heroku-postgresql", "as": "DATABASE" }],
+  "env": {
+    "PHX_SERVER": "true",
+    "PHX_HOST": {
+      "description": "Host of the website.",
+      "value": "localhost"
```
Comment (loicknuchel) [inline app.json:19]: `{{app-name}}.herokuapp.com`

File: README.md [hunk 2]
```diff
@@ -50,6 +50,19 @@ Azimutt goal is to be your ultimate tool to understand your database.
...
+
+You can use our Heroku template which includes Azimutt web app and a postgres database.
+
+[![Deploy](https://www.herokucdn.com/deploy/button.svg)](https://www.heroku.com/deploy)
+
+### Configuration
+
+Required: After the first deploy, please change the `PHX_HOST` config var to the domain assigned by Heroku or with your custom domain.
+
+Optional: Since Heroku does not allow exposing two different web services on the same dyno, our template uses the Azimutt default gateway [https://gateway.azimutt.app](https://gateway.azimutt.app).
```
Comment (loicknuchel) [inline README.md:63]: Remove gateway doc

File: README.md [hunk 3]
```diff
@@ -50,6 +50,19 @@ Azimutt goal is to be your ultimate tool to understand your database.
...
 
+## Deploy on Heroku
+
+You can use our Heroku template which includes Azimutt web app and a postgres database.
+
+[![Deploy](https://www.herokucdn.com/deploy/button.svg)](https://www.heroku.com/deploy)
+
+### Configuration
+
+Required: After the first deploy, please change the `PHX_HOST` config var to the domain assigned by Heroku or with your custom domain.
```
Comment (loicknuchel) [inline README.md:61]: Move to Deploy to Heroku section

File: app.json [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  "logo": "https://azimutt.app/images/logo_dark.svg",
+  "keywords": [
+    "erd",
+    "entity-relationship diagram",
+    "database tool",
+    "database schema",
+    "diagram"
+  ],
+  "stack": "container",
+  "addons": [{ "plan": "heroku-postgresql", "as": "DATABASE" }],
```
Comment (loicknuchel) [inline app.json:14]: Add S3 addon + env variables

File: app.json [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+      "description": "Host of the website.",
+      "value": "localhost"
+    },
+    "PORT": "4000",
+    "SECRET_KEY_BASE": {
+      "generator": "secret"
+    },
+    "FILE_STORAGE_ADAPTER": "local",
+    "AUTH_PASSWORD": "true",
+    "DATABASE_ENABLE_SSL": "true"
```
Comment (loicknuchel) [inline app.json:27]: Could be nice to also have:

- `EMAIL_ADAPTER=mailgun` with mailgun addon and the necessary env var (MAILGUN_DOMAIN, MAILGUN_API_KEY, MAILGUN_BASE_URL)
- `SKIP_ONBOARDING_FUNNEL=true`: so they try if fast without onboarding ^^
- `SKIP_EMAIL_CONFIRMATION=true`: same, faster test and don't require sending emails to test ^^
- `PUBLIC_SITE=false`: so I don't have copy of Azimutt site everywhere (I should remove it from the repo at some point)

Same for coolify ;)
++

File: backend/config/runtime.exs [hunk 6]
```diff
@@ -81,8 +81,8 @@ case System.fetch_env!("FILE_STORAGE_ADAPTER") do
   "s3" ->
     IO.puts("Setup S3 file storage")
     s3_host = System.get_env("S3_HOST") || ""
-    s3_key_id = System.get_env("S3_KEY_ID") || ""
-    s3_key_secret = System.get_env("S3_KEY_SECRET") || ""
+    s3_key_id = System.get_env("S3_KEY_ID") || System.get_env("S3_ACCESSS_KEY") || ""
+    s3_key_secret = System.get_env("S3_KEY_SECRET") || System.get_env("S3_SECRET_KEY") || ""
```
Comment (tonylyjones) [inline backend/config/runtime.exs:85]: Heroku addons defaut suffixes. Did not find how to define our suffixes

Comment (loicknuchel) [reply backend/config/runtime.exs:85]: Would be great to have a comment in the code to explain this as well ;)

**Review by loicknuchel — APPROVED** [review]

File: app.json [hunk 7]
```diff
@@ -19,8 +19,8 @@
   "env": {
     "PHX_SERVER": "true",
     "PHX_HOST": {
-      "description": "{{app-name}}.herokuapp.com or custom domain",
-      "value": ""
+      "description": "URL of Azimutt app.",
+      "value": "$(app-name).herokuapp.com"
```
Comment (loicknuchel) [inline app.json:23]: :+1:

**Review by loicknuchel — APPROVED** [review]

**Pull request merged**